CMakeLists.txt

```diff
@@ -79,18 +80,18 @@ target_link_libraries(multiprocess PRIVATE CapnProto::kj-async)
 set_target_properties(multiprocess PROPERTIES
     PUBLIC_HEADER "${MP_PUBLIC_HEADERS}")
 install(TARGETS multiprocess EXPORT Multiprocess
-  ARCHIVE DESTINATION lib COMPONENT lib
-  PUBLIC_HEADER DESTINATION include/mp COMPONENT lib)
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT lib
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/mp COMPONENT lib)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/pkgconfig/libmultiprocess.pc"
-  DESTINATION "lib/pkgconfig" COMPONENT lib)
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig COMPONENT lib)
```

Comment:
In commit "Use `GNUInstallDirs` module" (54bd57fb3b064c7f3d556b015e8f64fd8d234c19)

I think this change is right, but it did seem to cause a problem with a depends build that fanquake is fixing in https://github.com/bitcoin/bitcoin/pull/28846 commit https://github.com/bitcoin/bitcoin/pull/28846/commits/156366f10aa38c612e26a1f93d8503786f3d3a34

On fedora, replacing `lib` with `${CMAKE_INSTALL_LIBDIR}` here causes the pkgconfig file to be installed in `lib64/` instead of `lib/`, which seems right and makes sense according to explanations given https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html and https://github.com/Kitware/CMake/blob/master/Modules/GNUInstallDirs.cmake

But this is incompatible with the depends build that hardcodes the `lib/` directory: https://github.com/bitcoin/bitcoin/blob/d752349029ec7a76f1fd440db2ec2e458d0f3c99/depends/config.site.in#L91, so hardcoding lib again in https://github.com/bitcoin/bitcoin/pull/28846 is needed after this change.

Reply:
> Right. The current Bitcoin Core depends build system will handle any other package built and installed with CMake in this flawed way.

The depends system definitely has a bug because will installing the library in "lib64" on some platforms, but look for it in "lib" regardless of the platform. But the approach isn't really flawed, is it? It seems reasonable for the depends system to just always install the library in "lib" and always look for it there.

Reply:
I'm fine with either approach. But in the light of the upcoming migration from Autotools to CMake, the forced installation to the `lib` subdirectory looks useless.